docs: clarify Routing.Type=custom as experimental (#11111)
ipfs/kubo: docs/config.md

Default: `auto`

Type: `optionalString` (`null`/missing means the default)

### `Routing.DelegatedRouters`

An array of URL hostnames for delegated routers to be queried in addition to the Amino DHT when `Routing.Type` is set to `auto` (default) or `autoclient`.
These endpoints must support the [Delegated Routing V1 HTTP API](https://specs.ipfs.tech/routing/http-routing-v1/).

The special value `"auto"` uses delegated routers from [AutoConf](#autoconf) when enabled.

> [!TIP]
> Delegated routing allows IPFS implementations to offload tasks like content routing, peer routing, and naming to a separate process or server while also benefiting from HTTP caching.
>
> One can run their own delegated router either by implementing the [Delegated Routing V1 HTTP API](https://specs.ipfs.tech/routing/http-routing-v1/) themselves, or by using [Someguy](https://github.com/ipfs/someguy), a turn-key implementation that proxies requests to other routing systems. A public utility instance of Someguy is hosted at [`https://delegated-ipfs.dev`](https://docs.ipfs.tech/concepts/public-utilities/#delegated-routing).

Default: `["auto"]`

> Type: `array[string]` (URLs or `"auto"`)

Reviewer: Can `auto` be combined with custom ones? For example, if I want to combine Mainnet + custom content routing?

    This is critical to maintain to not harm the network.
- The operations `ipfs stats dht` will default to showing information about the accelerated DHT client

> [!CAUTION]
> **`Routing.AcceleratedDHTClient` Caveats:**
>
> 1. Running the accelerated client likely will result in more resource consumption (connections, RAM, CPU, bandwidth)

> >    - Users that are limited in the number of parallel connections their machines/networks can perform will likely suffer

Reviewer 2: ```suggestion
>    - Users that are limited in the number of parallel connections their machines/networks can perform will be most affected
```

> #### `Routing.Routers: Parameters`

Reviewer: ```suggestion
#### `Routing.Routers.[router-name].Parameters`
```

For consistency and clarity
